app/src/main/java/org/oppia/app/home/topiclist/PromotedStoryViewModel.kt

```diff
@@ -0,0 +1,43 @@
+package org.oppia.app.home.topiclist
+
+import android.view.View
+import androidx.appcompat.app.AppCompatActivity
+import androidx.databinding.ObservableField
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.ViewModel
+import org.oppia.app.fragment.FragmentScope
+import org.oppia.app.home.HomeActivity
+import org.oppia.app.model.PromotedStory
+import org.oppia.app.viewmodel.ObservableViewModel
+import javax.inject.Inject
+
+// TODO(#283): Add download status information to promoted-story-card.
+
+/** [ViewModel] for displaying a promoted story. */
+@FragmentScope
+class PromotedStoryViewModel @Inject constructor(
```

Comment:
I suggest not making this injectable. I ran into a similar situation when rearranging the state fragment's recyclerview adapter, and I came to the conclusion that view models which are stored in recycler view adapter lists shouldn't be injected (since they ought to be constructed, instead). That will solve the need to depend on HomeActivity directly because you can pass in RouteToContinuePlayingListener in the view model's constructor to call into. That keeps the UI dependencies clean (members of the UI hierarchy should only directly call into children and children should indirectly call into parents).

Reply:
Done. Removed Inject and now PromotedStoryViewModel is directly calling `TopicActivity` and `ContinuePlayingActivity`.